system/mp/mpcomm.cpp

```diff
@@ -448,11 +448,85 @@ class CMPConnectThread: public Thread
     ISocket *listensock;
     CMPServer *parent;
     int mpSoMaxConn;
+    Owned<IThreadPool> threadPool;
+    unsigned acceptThreadPoolSize = 0;
+
     Owned<IAllowListHandler> allowListCallback;
     void checkSelfDestruct(void *p,size32_t sz);
 
     Owned<ISecureSocketContext> secureContextServer;
 
+    class CSlowClientProcessor : implements IThreaded
+    {
+        CMPConnectThread &owner;
+        CThreaded threaded;
+        std::vector<Owned<ISocket>> slowClientsSocks;
+        CriticalSection crit;
+        Semaphore sem;
+        std::atomic<bool> stopped = true;
+
+        void clearSocks()
+        {
+            CriticalBlock b(crit);
+            for (auto &sock : slowClientsSocks)
+                sock->close();
```

Comment:
reading up, I think it is unsafe, also, the handling of getting slowClientsSocks.back() in crit, then later calling pop_back() is unsafe is new items are being added.

I am going to remove the socket from the slowClientsSocks before calling handleAcceptedSocket.
It will mean it can't be interrupted and will timeout, but that's not really a problem, just a theoretical slightly longer delay during shutdown if happens to be on a stalling socket.